clash-protocols/src/Protocols/PacketStream/Routing.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_HADDOCK hide #-}
+
+{- |
+Copyright  :  (C) 2024, QBayLogic B.V.
+License    :  BSD2 (see the file LICENSE)
+Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+
+Provides a packet arbiter and dispatcher, for merging and splitting packet streams.
+-}
+module Protocols.PacketStream.Routing (
+  packetArbiterC,
+  ArbiterMode (..),
+  packetDispatcherC,
+  routeBy,
+) where
+
+import Clash.Prelude
+
+import Protocols
+import Protocols.PacketStream.Base
+
+import qualified Data.Bifunctor as B
+import Data.Maybe
+
+-- | Collect mode for `packetArbiterC`
+data ArbiterMode
+  = -- | Collect in a round-robin fashion. Fair and cheaper than `Parallel`.
+    RoundRobin
+  | -- | Check components in parallel. This mode has a higher throughput, but is
+    -- biased towards the last source and also slightly more expensive.
+    Parallel
```

Comment:
Just checked it, it's not quite the same. `roundrobinCollect` also has a `NoSkip` mode:
```
Collect in a roundrobin fashion. If a component does not produce data, wait until it does.
```
We could handle add this mode to the arbiter as well.

Reply:
That would be nice, let's try to make it uniform

Reply:
Done. Added a fat warning to the `NoSkip` mode because it can easily lead to starvation. In fact, the test can only successfully run if the same amount of packets are generated for each source.